client/src/main/java/org/eclipse/hono/client/impl/DelegatedCommandSenderImpl.java

```diff
@@ -183,7 +183,7 @@ public String getEndpoint() {
         final String targetAddress = getTargetAddress(tenantId, deviceId);
         final String replyToAddress = command.isOneWay() ? null
                 : String.format("%s/%s/%s",
-                CommandConstants.COMMAND_ENDPOINT, tenantId, command.getReplyToId());
+                CommandConstants.NORTHBOUND_COMMAND_RESPONSE_ENDPOINT, tenantId, command.getReplyToId());
```

Comment:
a comment explaining why this is *not* the legacy endpoint would be helpful ...

Reply:
Thanks for pointing at that line. Always choosing the `NORTHBOUND_COMMAND_RESPONSE_ENDPOINT` was actually wrong here. I've added and used a `Command.getReplyToEndpoint` method here.